src/comm/LinkManager.cc

```diff
@@ -478,8 +479,28 @@ void LinkManager::_updateAutoConnectLinks(void)
         udpConfig->setDynamic(true);
         SharedLinkConfigurationPointer config = addConfiguration(udpConfig);
         createConnectedLink(config);
+
         emit linkConfigurationsChanged();
     }
+#ifndef __android__
+    if (_autoConnectSettings->autoConnectNmeaPort()->cookedValueString() == "UDP Port")
+    {
+        if (_nmeaSocket == nullptr)
+        {
+            qCDebug(LinkManagerLog) << "Creating UDP socket for NMEA stream";
+            _nmeaSocket = new UdpIODevice(this);
```

Comment:
it's much easier to use a unique smart pointer than checking for nullptr and using new all the place.